sgv2-service-common/src/main/java/io/stargate/sgv2/common/grpc/DefaultStargateBridgeClient.java

```diff
@@ -107,20 +114,55 @@ private CqlKeyspaceDescribe getAuthorizedKeyspace(String keyspaceName) {
     List<Boolean> authorizations =
         authorizeSchemaReads(
             names.stream()
-                .map(n -> SchemaReads.keyspace(n, SchemaRead.SourceApi.REST))
+                .map(n -> SchemaReads.keyspace(n, sourceApi))
                 .collect(Collectors.toList()));
     List<CqlKeyspaceDescribe> keyspaces = new ArrayList<>(names.size());
     int i = 0;
     for (String name : names) {
-      CqlKeyspaceDescribe ks;
-      // Null check because it could have been deleted since we listed the names
-      if (authorizations.get(i) && (ks = getAuthorizedKeyspace(name)) != null) {
-        keyspaces.add(ks);
+      if (authorizations.get(i)) {
+        getAuthorizedKeyspace(name).ifPresent(keyspaces::add);
       }
     }
     return keyspaces;
   }
 
+  @Override
+  public Optional<Schema.CqlTable> getTable(String keyspaceName, String tableName)
+      throws UnauthorizedTableException {
+    if (!authorizeSchemaRead(SchemaReads.table(keyspaceName, tableName, sourceApi))) {
+      throw new UnauthorizedTableException(keyspaceName, tableName);
+    }
+    return getAuthorizedKeyspace(keyspaceName)
+        .flatMap(
+            keyspace ->
+                keyspace.getTablesList().stream()
+                    .filter(t -> t.getName().equals(tableName))
+                    .findFirst());
+  }
+
+  @Override
+  public List<Schema.CqlTable> getTables(String keyspaceName) {
+    return getAuthorizedKeyspace(keyspaceName)
+        .map(
+            keyspace -> {
+              List<Schema.CqlTable> tables = keyspace.getTablesList();
+              List<Boolean> authorizations =
+                  authorizeSchemaReads(
+                      tables.stream()
+                          .map(t -> SchemaReads.table(keyspaceName, t.getName(), sourceApi))
+                          .collect(Collectors.toList()));
+              List<Schema.CqlTable> authorizedTables = new ArrayList<>(tables.size());
+              int i = 0;
```

Comment:
here also, `int i = 0` is not changed or?

Reply:
Do we have a unit test for the `DefaultStargateBridgeClient`

Reply:
I'll add the test in a follow-up PR, it's going to take a bit longer than expected and I don't want to hold other work that depends on this PR.